frontend/lib/src/components/shared/Dropdown/VirtualDropdown.tsx

```diff
@@ -120,9 +120,11 @@ const VirtualDropdown = React.forwardRef<any, any>((props, ref) => {
         height={height}
         itemCount={children.length}
         itemData={children}
-        itemKey={(index: number, data: { props: OptionListProps }[]) =>
-          data[index].props.item.value
-        }
+        itemKey={(index: number, data: { props: OptionListProps }[]) => {
+          const { id, value } = data[index].props.item
+
+          return id ?? value
```

Comment:
nit: in which cases can the `id` not be set?

Reply:
It should always be defined for use cases, but Virtual Dropdown is generic enough that I want to make sure it always works, so having the value as a fallback makes sense.